rules/aip0158/response_repeated_first_field.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0158
+
+import (
+	"fmt"
+
+	"github.com/googleapis/api-linter/lint"
+	"github.com/jhump/protoreflect/desc"
+)
+
+var responseRepeatedFirstField = &lint.MessageRule{
+	Name:   lint.NewRuleName(158, "response-repeated-first-field"),
+	OnlyIf: isPaginatedResponseMessage,
+	LintMessage: func(m *desc.MessageDescriptor) []lint.Problem {
+		// Rule check: for all messages that end in Response and contain a next_page_token field.
+		// Throw a linter warning if the first field in the message (by both position and field number) is not repeated
+		if m.FindFieldByName("next_page_token") == nil {
+			return nil
+		}
+
+		if !m.FindFieldByNumber(1).IsRepeated() {
```

Comment:
Can you combine checking first field by number and by position into one? Like:
```golang
// check if the message is empty
if len(m.GetFields()) == 0 {
 ...
}

firstField := m.FindFieldByNumber(1)
if firstField == nil {
  firstField := m.GetFields()[0]
}

if !firstField.IsRepeated() {
  ...
}

Reply:
If we combine it, we will miss this case.
```
message Profile {
  string name = 1;
}
message ListStudentProfilesResponse {
  Profile student_profiles = 2;
  repeated string next_page_token = 1;
}
` ``

Reply:
I see, thank you!